dashboard/src/components/AppView/AppView.tsx

```diff
@@ -126,22 +126,26 @@ class AppView extends React.Component<IAppViewProps, IAppViewState> {
 
     const sockets: WebSocket[] = [];
     manifest.forEach(i => {
+      const id = `${i.kind}/${i.metadata.name}`;
       switch (i.kind) {
         case "Deployment":
+          this.setState({ deployments: { ...this.state.deployments, [id]: { isFetching: true } } });
```

Comment:
Why don't you initialize the state with isFetching: true and remove these lines? 

Also, does this line ever set deployments? It seems that the isFetching thing is what is relevant isn't it?

Reply:
> Why don't you initialize the state with isFetching: true and remove these lines?

I am not sure if I understand this comment. Note that that the `isFetching` property depends on the independent elements of the array, I cannot initialize `isFetching` property.

> Also, does this line ever set deployments? It seems that the isFetching thing is what is relevant isn't it?

True, I am only interested in the isFetching property